File: src/main/java/io/golos/commun4J/model/VoteRequest.java
package io.golos.commun4J.model;

import com.memtrip.eos.abi.writer.Abi;
import com.memtrip.eos.abi.writer.IntCompress;
import com.memtrip.eos.abi.writer.NameCompress;
import com.memtrip.eos.abi.writer.StringCompress;

@Abi
public class VoteRequest {
    private CommunName name;
    private CommunName author;
    private String permlink;
    private short weight;

    public VoteRequest(CommunName name, CommunName author, String permlink, short weight) {
        this.name = name;
        this.author = author;
        this.permlink = permlink;
        this.weight = weight;
    }

    @NameCompress
    public String getName() {
        return name.getName();
    }

    @NameCompress
    public String getAuthor() {
        return author.getName();
    }

    @StringCompress
    public String getPermlink() {
        return permlink;
    }

    @IntCompress
    public short getWeight() {
        return weight;
    }


    @Override
    public String toString() {
        return "VoteRequest{" +
                "name=" + name +
                ", author=" + author +
                ", permlink='" + permlink + '\'' +
                ", weight=" + weight +
                '}';
    }
}

File: src/main/java/io/golos/commun4J/model/VoteRequest.kt
package io.golos.commun4J.model

import com.memtrip.eos.abi.writer.Abi
import com.memtrip.eos.abi.writer.ChildCompress
import com.memtrip.eos.abi.writer.IntCompress
import com.memtrip.eos.abi.writer.NameCompress

@Abi
internal class VoteRequest(private val voter: CommunName,
                           private val discussionId: DiscussionId,
                           private val weight: Short) {

    val getVoter: String
        @NameCompress get() = voter.name

    val getDiscussionId
        @ChildCompress get() = discussionId


    val getWeight: Int
        @IntCompress get() = weight.toInt()

    override fun toString(): String {
        return "VoteRequest(voter=$voter, discussionId=$discussionId, weight=$weight)"
    }


}
